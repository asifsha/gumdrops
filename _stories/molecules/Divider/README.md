The standard  `<Divider>` component is used to separate distinct segments of content.

If the divider is collapsible, pass in a callback function that sets the state, and include the state in your component.

**Example**:
```
state {
    open: false
}

const toggle = () => {
    this.setState({ open: !this.state.open });
};

return(
    <div>
        <Divider
            label="Collapsible Label"
            collapsible
            open={ this.state.open }
            callback= { this.toggle }
        />
        this.state.open && <div>Content to show when toggled open</div>
    </div>
)
```
