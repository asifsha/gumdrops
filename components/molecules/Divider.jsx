import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

const Divider = ({
    label,
    centered = false,
    collapsible = false,
    open,
    callback,
    className,
    style
}) => {
    const baseClass = 'gds-divider';
    const rootClass = cx(baseClass, className, {
        [`${baseClass}--${centered}`]: centered,
        [`${baseClass}--${collapsible}`]: collapsible
    });

    const arrowClasses = cx(`${baseClass}__arrow`, {
        [`${baseClass}__arrow--collapse`]: !open
    });

    return (
        <div className={rootClass} style={style} onClick={callback}>
            {centered && <span className="gds-divider__line" />}
            <span className="gds-divider__label gds-form-group__label">{label}</span>
            <span className="gds-divider__line" />
            {collapsible && <span className={arrowClasses} />}
        </div>
    );
};

Divider.displayName = 'Divider';

Divider.defaultProps = {
    label: null,
    centered: false,
    collapsible: false,
    open: null,
    callback: null,
    style: {}
};

Divider.propTypes = {
    label: PropTypes.string,
    centered: PropTypes.bool,
    collapsible: PropTypes.bool,
    open: PropTypes.bool,
    callback: PropTypes.func,
    className: PropTypes.string,
    style: PropTypes.object
};

export default Divider;
