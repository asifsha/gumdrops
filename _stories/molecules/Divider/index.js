import React from 'react';
import { boolean, text, object } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

import readme from './README.md';
import Divider from '../../../components/molecules/Divider';

const component = () => (
    <Divider
        label={text('Label', 'Default Divider')}
        centered={boolean('Centered', false)}
        collapsible={boolean('Collapsible', false)}
        open={boolean('Open', false)}
        className={text('ClassName', '')}
        style={object('Style', {})}
        callback={action('divider_toggled')}
    />
);

export default [readme, component];
