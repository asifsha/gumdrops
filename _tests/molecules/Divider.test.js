/* globals mount */
import React from 'react';
import Divider from '../../components/molecules/Divider';

const defaultProps = {
    label: 'Divider Label',
    centered: false,
    collapsible: false,
    open: false,
    callback: () => {},
    style: { width: 150 }
};

describe('Expect <Divider>', () => {
    it('to render', () => {
        const wrapper = mount(<Divider {...defaultProps} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('to render open', () => {
        const props = {
            ...defaultProps,
            collapsible: false,
            open: true
        };
        const wrapper = mount(<Divider {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
