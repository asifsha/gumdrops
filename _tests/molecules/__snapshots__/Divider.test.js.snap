// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <Divider> to render 1`] = `
<Divider
  callback={[Function]}
  centered={false}
  collapsible={false}
  label="Divider Label"
  open={false}
  style={
    Object {
      "width": 150,
    }
  }
>
  <div
    className="gds-divider"
    onClick={[Function]}
    style={
      Object {
        "width": 150,
      }
    }
  >
    <span
      className="gds-divider__label gds-form-group__label"
    >
      Divider Label
    </span>
    <span
      className="gds-divider__line"
    />
  </div>
</Divider>
`;

exports[`Expect <Divider> to render open 1`] = `
<Divider
  callback={[Function]}
  centered={false}
  collapsible={false}
  label="Divider Label"
  open={true}
  style={
    Object {
      "width": 150,
    }
  }
>
  <div
    className="gds-divider"
    onClick={[Function]}
    style={
      Object {
        "width": 150,
      }
    }
  >
    <span
      className="gds-divider__label gds-form-group__label"
    >
      Divider Label
    </span>
    <span
      className="gds-divider__line"
    />
  </div>
</Divider>
`;
